.circleci/config.yml

```diff
@@ -16,26 +16,61 @@ jobs:
       - checkout
 
       - restore_cache:
-          key: oppia-tools-cache
+          keys:
+            - v1.0-oppia-tools-cache-{{ checksum "oppia-tools-dependency-list.txt" }}
```

Comment:
Hmm, where does "v1.0" come from, and are we maintaining any sort of formal version number? If so I think we should also have a written procedure for when to update the version number and how to do it (which parts of the code need to be changed etc.).

Also we should clarify the scope of this version number (is it for the oppia-tools dir or the broader site or something else).

Same question, everywhere else we have versioning.

(Also I wonder if the checksum alone suffices.)

Reply:
1) The cache versioning idea comes from CircleCI doc:
https://circleci.com/docs/2.0/caching/#using-keys-and-templates
> Tip: Caches are immutable so it is useful to start all your cache keys with a version prefix, for example v1-.... This enables you to regenerate all of your caches by incrementing the version in this prefix.

I am not sure how to proceed here.

On one hand, I think I can do without this cache number thing? On the other hand, I feel like there must be a reason for CircleCI to recommend me doing this. I am split here.

We can include comments on how to best scope this version number. Specifically, this version number is only used to control the cache version.
2) I have added `branch-name on top of `checksum`. Would that suffice? What do you think is missing here?

Reply:
1. Perhaps you could ask them? Seems like it's worth getting clarity.

2. I'm not really sure where this comment comes from -- I don't remember referring to the branch name. Are you referring to the "scope" comment? If so, what I actually meant there was, is "v1.0" referring to the version of the application, or the version of the particular dependency, etc. -- it wasn't clear.